src/pages/dar_application/ProgressReportApplication.tsx

```diff
@@ -19,10 +33,23 @@ export const ProgressReportApplication = ({dar, readOnlyMode=true}: ProgressRepo
                     <h4>Intellectual Property Summary</h4>
                     {dar?.intellectualPropertySummary ?? "PLACEHOLDER Intellectual Property Summary"}
                 </div>
-                {dar?.parentId && <div>
+                <div data-cy='remove-datasets'>
+                    <div className='progress-report-step-card'>
+                        <h2>Step 3: Dataset(s) in this DAR</h2>
+                        <p style={{ marginBottom: '1rem' }}>Currently selected datasets:</p>
+                        <SelectableDatasets
+                            disabled={readOnlyMode}
+                            datasets={datasets}
+                            setSelectedDatasets={(selectedDatasets) => {
+                                setFormState({...formState, datasetIds: selectedDatasets.map((ds) => ds.datasetId)})}
```

Comment:
```suggestion
                                setFormState((prevState) => ({...prevState, datasetIds: selectedDatasets.map((ds) => ds.datasetId) }));
```

It looks like we need to use an "updater" function to use previous state to update the new state - https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

I _think_ this was the actual source of our infinite looping in the useEffect.

Reply:
ohhh that makes sense! thank you! Do you think I should make this change in place of what we did yesterday?